### PR TITLE
Convert COS job to SemeioScript and use reporter for exporting data

### LINKED ISSUE #45: [COS] Output singular values
Correlated Observations scaling

Users would like output of singular values to stdout

https://github.com/equinor/semeio/blob/2e779f8d4b3c52bafbc519796dbe5f928ea57af7/semeio/jobs/correlated_observations_scaling/scaled_matrix.py#L81

### LINKED ISSUE #112: The COS job should publish data
COS should utilise the new reporter to publish data for later inspection. In particular, it should make available:
- singular values (previously #45)

### PR DESCRIPTION
**Pull request by ManInFez**
Resolves #112 

Use new reporter in semeio.communications for `reporting` data.

### CODE REVIEW

File: semeio/jobs/correlated_observations_scaling/job.py [hunk 1]
```diff
@@ -35,6 +37,20 @@ def scale(self, measured_data):
         matrix.std_normalization(inplace=True)
 
         scale_factor = matrix.get_scaling_factor(config.CALCULATE_KEYS)
+        events = config.CALCULATE_KEYS
+        data_matrix = matrix.get_data_matrix()
+        nr_components, singular_values = DataMatrix.get_nr_primary_components(
+            data_matrix, threshold=events.threshold
+        )
+        self._reporter.publish("COS.svd", list(singular_values))
```
Comment (markusdregi) [inline semeio/jobs/correlated_observations_scaling/job.py:45]: The reporter is already namespace with the name of the workflow; should we perhaps do `svd` only?

File: tests/communication/test_integration.py [hunk 2]
```diff
@@ -40,7 +41,9 @@ def test_semeio_script_integration(tmpdir):
     assert list(range(10)) == reported_data
 
     # Assert that logs were forwarded correctly
-    with open("storage/reports/TestWorkflowJob/log.txt") as f:
+    with open(
+        os.path.join("storage/reports/TestWorkflowJob/", SEMEIOSCRIPT_LOG_FILE)
```
Comment (markusdregi) [inline tests/communication/test_integration.py:45]: Perhaps we can store the filename in a variable? I agree with the black formatting, but at some point a variable makes the whole thing more readable...

File: tests/jobs/correlated_observations_scaling/unit/test_scaling_job.py [hunk 3]
```diff
@@ -64,12 +65,14 @@ def test_valid_job(calc_key, app_key, obs_keys, obs_with_data, expect_valid, err
         "CALCULATE_KEYS": {"keys": [{"key": calc_key}]},
         "UPDATE_KEYS": {"keys": [{"key": app_key}]},
     }
+
+    reporter = FileReporter(os.path.realpath(os.getcwd()))
```
Comment (markusdregi) [inline tests/jobs/correlated_observations_scaling/unit/test_scaling_job.py:69]: Would be nice with a test loading the published data again...

File: semeio/jobs/correlated_observations_scaling/scaled_matrix.py [hunk 4]
```diff
@@ -47,20 +48,20 @@ def get_scaling_factor(self, events):
...
 
-        return self._calculate_scaling_factor(nr_observations, nr_components)
+        return DataMatrix.calculate_scaling_factor(nr_observations, nr_components)
 
     def _get_data(self):
         return self.data[~self.data.index.isin(["OBS", "STD"])]
 
     @staticmethod
-    def _get_nr_primary_components(data_matrix, threshold):
+    def get_nr_primary_components(data_matrix, threshold):
```
Comment (markusdregi) [inline semeio/jobs/correlated_observations_scaling/scaled_matrix.py:64]: I don't really see why this should be a static method 🤔

Comment (markusdregi) [reply semeio/jobs/correlated_observations_scaling/scaled_matrix.py:64]: Passing in `self` would be more in line with `get_scaling_factor` and others.

Comment (markusdregi) [reply semeio/jobs/correlated_observations_scaling/scaled_matrix.py:64]: Also, `get_nr_primary_components`, `calculate_scaling_factor` and `std_normalization` is not the most coherent API I've ever seen 😂 Should we try to align a bit?

**Comment by ManInFez** [comment]
> Premature approval
Dont worry, it happens to everyone...

File: tests/jobs/correlated_observations_scaling/unit/test_scaling_job.py (hunk 3 above)
Comment (ManInFez) [reply tests/jobs/correlated_observations_scaling/unit/test_scaling_job.py:69]: I agree. I have moved it to the integration test though, as the scale functions are not run in the unit test

**Pull request merged**